torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -146,6 +149,10 @@ int64_t DistAutogradContainer::getMaxId() {
   return max_id_;
 }
 
+void DistAutogradContainer::set_current_context_id(int64_t context_id) {
```

Comment:
Should we add a `TORCH_INTERNAL_ASSERT` here such that the current_context_id_ isn't set already?